neqo-transport/src/path.rs

```diff
@@ -152,13 +110,14 @@ impl Paths {
     }
 
     fn retire(to_retire: &mut Vec<u64>, retired: &PathRef) {
-        let seqno = retired
-            .borrow()
-            .remote_cid
-            .as_ref()
-            .unwrap()
-            .sequence_number();
-        to_retire.push(seqno);
+        if let Some(cid) = &retired.borrow().remote_cid {
+            let seqno = cid.sequence_number();
+            if cid.connection_id().len() == 0 {
```

Comment:
```suggestion
            if cid.connection_id().is_empty() {
```

Would the new `ConnectionId::is_empty()` function work here as well?